### ISSUE TITLE
Single parameter async + arrow function

### ISSUE THREAD

**Issue by felixsanz** [eslint/eslint#5489]
ESLint 2.3.0 using babel-eslint.

On async functions that are also arrow functions and have a single parameter, ESLint doesn't complain about parentheses. Example:

```javascript
app.use(async (ctx) => {
// code
})
```

ESLint should complain like when you use a normal function. Expecting:

```javascript
app.use(async ctx => {
// code
})
```

**Comment by eslintbot** [comment]
Hi @felixsanz, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by alberto** [comment]
Do you mean in conjunction with `arrow-parens: as-needed`?

**Comment by alberto** [comment]
Also, I don't think we support `async` directly since it's experimental.

**Comment by felixsanz** [comment]
@alberto  so is this a babel-eslint issue?

**Comment by ilyavolodin** [comment]
@felixsanz If it works without `async` as expected, then it's an issue for `eslint-plugin-babel` most likely, or possibly `babel-eslint`.
